### PR TITLE
TST: add test for rolling max with DatetimeIndex

### LINKED ISSUE #21096: regression in 0.23.0 on rolling max with DatetimeIndex.
Behaviour of .rolling() with datetimeindex appears to have changed or regressed between 0.22 & 0.23 versions of pandas.

See below for cut-down test case exhibiting behaviour.

Also see below for different output produced on same code on same data with 2 different pandas versions.

```python
import numpy as np
import pandas as pd


def main():
    print('pandas', pd.__version__)
    n = 10
    index = pd.date_range(
        start='2018-1-1 01:00:00',
        freq='1min',
        periods=n,
    )
    s = pd.Series(
        data=0,
        index=index,
    )
    s.iloc[1] = np.nan
    s.iloc[-1] = 2
    print(s)
    maxes = s.rolling(window=f'{n}min').max()
    print(maxes.value_counts(dropna=False))


if __name__ == '__main__':
    main()
```

ACTUAL OUTPUT
```
pandas 0.22.0      
2018-01-01 01:00:00    0.0             
2018-01-01 01:01:00    NaN             
2018-01-01 01:02:00    0.0             
2018-01-01 01:03:00    0.0             
2018-01-01 01:04:00    0.0             
2018-01-01 01:05:00    0.0             
2018-01-01 01:06:00    0.0             
2018-01-01 01:07:00    0.0             
2018-01-01 01:08:00    0.0             
2018-01-01 01:09:00    2.0             
Freq: T, dtype: float64                
0.0    9           
2.0    1           
dtype: int64       
pandas 0.23.0      
2018-01-01 01:00:00    0.0             
2018-01-01 01:01:00    NaN             
2018-01-01 01:02:00    0.0             
2018-01-01 01:03:00    0.0             
2018-01-01 01:04:00    0.0             
2018-01-01 01:05:00    0.0             
2018-01-01 01:06:00    0.0             
2018-01-01 01:07:00    0.0             
2018-01-01 01:08:00    0.0             
2018-01-01 01:09:00    2.0             
Freq: T, dtype: float64                
0.0    10          
dtype: int64       

```

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #21096
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/window/test_moments.py [hunk 1]
```diff
@@ -1891,6 +1891,20 @@ def test_rolling_corr_with_zero_variance(self, window):
 
         assert s.rolling(window=window).corr(other=other).isna().all()
 
+    def test_rolling_max_datetimeindex(self):
```
Comment (jreback) [inline pandas/tests/window/test_moments.py]: move to tests/window/test_timeseries_window.py and put near similar tests (this might be a duplicate)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/window/test_moments.py (hunk 1 above)
Comment (ganevgv) [reply pandas/tests/window/test_moments.py]: Thanks for the feedback, @jreback. Moved the test to `tests/window/test_timeseries_window.py`. This test is different by being the only one using minutes as frequency in the file. Other minor differences is that it's using Series as opposed to DataFrame and explicit definition of expected.

Having said that, I agree that there are very similar tests and I'm happy to remove it all together.

Comment (WillAyd) [reply pandas/tests/window/test_moments.py]: @ganevgv if you would like to clean that up and parametrize frequencies / containers that would be *very* welcome

**Review by WillAyd — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ganevgv 

I would love if you could parametrize some of these tests (e.g. also test min) in a followup if you can.

**Comment by ganevgv** [comment]
Thanks for the feedback @jreback, @WillAyd - created a new PR using parametrize over different operations and frequencies - #29932

**Pull request merged**